src/libical/icaltypes.c

```diff
@@ -130,7 +130,18 @@ struct icalreqstattype icalreqstattype_from_string(const char *str)
 
     p2 = strchr(p1 + 1, ';');
     if (p2 != 0 && *p2 != 0) {
+#if 0
         stat.debug = icalmemory_tmp_copy(p2 + 1);
+#else
+        // Don't allow (fuzzer) garbage else somethine in icalmemory fails
+        char *s = (char *)(p2 + 1);
+        while (*s && isprint((unsigned char)*s)) {
```

Comment:
What about moving this up to the top of the function and failing immediately if it encounters garbage in `str`? We're reading from `str` higher up, so seems like we should be checking earlier too.